modules/core/src/main/java/org/apache/ignite/events/NodeValidatedEvent.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.events;
+
+import org.apache.ignite.cluster.ClusterNode;
+import org.apache.ignite.spi.IgniteNodeValidationResult;
+
+/**
+ *  This event is triggered if the joining node validation complete.
+ *
+ * @see EventType#EVT_NODE_VALIDATION_FAILED
+ */
+public class NodeValidatedEvent extends EventAdapter {
+    /** */
+    private static final long serialVersionUID = 0L;
+
+    /** The node that attempted to join cluster. */
+    private ClusterNode evtNode;
+
+    /** Validation result. */
+    private IgniteNodeValidationResult validationRes;
+
+    /**
+     * Creates new node validation event with given parameters.
+     *
+     * @param node Local node.
+     * @param msg Optional event message.
+     * @param type Event type.
```

Comment:
Lets do event type as const ang get it from `EventType`. This class is for only one type of event.